src/librustc/middle/traits/coherence.rs

```diff
@@ -178,9 +178,9 @@ fn ty_is_local_constructor<'tcx>(tcx: &ty::ctxt<'tcx>, ty: Ty<'tcx>) -> bool {
             false
         }
 
-        ty::ty_enum(def_id, _) |
-        ty::ty_struct(def_id, _) => {
-            def_id.krate == ast::LOCAL_CRATE
+        ty::ty_enum(def, _) |
+        ty::ty_struct(def, _) => {
+            def.def_id.krate == ast::LOCAL_CRATE
```

Comment:
Makes me wonder if we should have an `is_local()` method on `DefId` so we can do `def.def_id.is_local()`.